src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -11,23 +11,25 @@ use std::process::Command;
 use uuid::Uuid;
 
 use devicemapper as dm;
-use devicemapper::{DM, DataBlocks, DmDevice, DmName, DmNameBuf, IEC, LinearDev, MetaBlocks,
-                   Sectors, ThinDev, ThinDevId, ThinPoolDev, ThinPoolStatusSummary, device_exists};
+use devicemapper::{DM, DataBlocks, Device, DmDevice, DmName, DmNameBuf, IEC, LinearDev,
+                   LinearDevTargetParams, LinearTargetParams, MetaBlocks, Sectors, TargetLine,
+                   ThinDev, ThinDevId, ThinPoolDev, ThinPoolStatusSummary, device_exists};
 
 use super::super::super::engine::Filesystem;
 use super::super::super::errors::{EngineError, EngineResult, ErrorEnum};
 use super::super::super::structures::Table;
-use super::super::super::types::{DevUuid, FilesystemUuid, Name, PoolUuid, RenameAction};
+use super::super::super::types::{FilesystemUuid, Name, PoolUuid, RenameAction};
 
-use super::super::blockdevmgr::{BlkDevSegment, BlockDevMgr, Segment, map_to_dm};
+use super::super::backstore::Backstore;
 use super::super::device::wipe_sectors;
 use super::super::devlinks;
+use super::super::dmnames::{FlexRole, ThinPoolRole, ThinRole, format_flex_name,
+                            format_thinpool_name, format_thin_name};
```

Comment:
Please sort `FlexRole, ThinPoolRole, ThinRole, format_flex_name, format_thin_name, format_thinpool_name`

Reply:
ASCII code for "_" precedies code for "p". Should stay as is.

Reply:
@mulkieran Yes `_` (95d) comes before `p` (112d), thus `format_thin_name` comes before `format_thinpool_name` in ASCII sorting.

As I stated in a previous PR I'm not doing these checks by hand anymore.  If we don't like ASCII sort order we can change it, but we need to pick one and stick with it as the tool I wrote will keep flagging an error if `format_thinpool_name` comes before `format_thin_name`.

Reply:
@tasleson When will your tool be mature enough to include in our CI?

Reply:
@mulkieran After it gets some use from multiple users and we find the results meet our expectations.

Reply:
@tasleson What are the expectations that you think it doesn't meet yet?